mdfreader/mdf4reader.py

```diff
@@ -1312,13 +1324,30 @@ def read4(self, file_name=None, info=None, multi_processed=False, channel_list=N
         else:
             channel_set_file = None
 
+        # if source_list is not None:
+        #     source_set_file = set(source_list)
+
         # Read information block from file
-        if info is None:
-            if self.info is None:
-                info = Info4(self.fileName, None,
-                             filter_channel_names=filter_channel_names, minimal=minimal)
-            else:
-                info = self.info
+        # If the source list is present, the information block from file are organised
```

Comment:
I do not understand this change, you can pass over multi_sources parameter being None or some proper content

Reply:
Upon reviewing the code again, I realized that I wrote it hastily. I have revised this section as follows:

if self.info is None:
    if source_list is None:
        multi_sources = False
    else:
        # Extract the information block from the file and arrange it based on the provided source list.
        # This organization enables us to associate each signal with its respective source message,
        # indicating the device of origin.

        multi_sources = True

    info = Info4(self.fileName, None,
                 filter_channel_names=filter_channel_names, minimal=minimal,
                 multi_sources=multi_sources)
else:
    info = self.info

In this revision, I added a has_source_list flag to clearly indicate whether a source list is present. If you have any further suggestions or if there's anything else you'd like me to address, please feel free to let me know. I appreciate your feedback.